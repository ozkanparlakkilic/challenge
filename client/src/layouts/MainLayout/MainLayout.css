.main-layout {
    overflow-y: hidden;
}
