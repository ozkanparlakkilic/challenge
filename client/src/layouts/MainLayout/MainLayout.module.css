.main_layout {
    overflow-y: hidden;
}
