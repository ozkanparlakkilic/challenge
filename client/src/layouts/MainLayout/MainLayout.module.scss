@import '@/styles/scss/_variables';
@import '@/styles/scss/_mixins';

.main_layout {
    @include flex-1;
    overflow-y: $overflowY;
}