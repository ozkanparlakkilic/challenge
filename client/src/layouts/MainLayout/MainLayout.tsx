import React, { useEffect } from 'react';
import { useAppDispatch } from '../../store/hooks';
import { setUserData } from '../../store/slices/userSlice';
import styles from './MainLayout.module.css';

interface ChildrenProps {
    children: React.ReactNode;
}

const MainLayout = ({ children }: ChildrenProps) => {

    const dispatch = useAppDispatch()

    useEffect(() => {
        dispatch(setUserData())
    },[dispatch])

    return <div className={`flex-1 ${styles.main_layout}`}>{children}</div>;
};

export default MainLayout;
