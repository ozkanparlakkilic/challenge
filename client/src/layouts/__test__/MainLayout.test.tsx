import React from 'react';
import { render } from '@testing-library/react';
import MainLayout from '../MainLayout/MainLayout';

describe('Renderings', () => {
    it('should render the child component', () => {
      const childElement = ("<div>Testing</div>")
      const { container } = render(<MainLayout children={childElement} />)
      expect(container).toMatchSnapshot();
    });
});