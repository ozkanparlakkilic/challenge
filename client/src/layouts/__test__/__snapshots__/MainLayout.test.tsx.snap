// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderings should render the child component 1`] = `
<div>
  <div
    class="flex-1 main_layout"
  >
    &lt;div&gt;Testing&lt;/div&gt;
  </div>
</div>
`;
